auslib/web/admin/swagger/api.yaml

```diff
@@ -168,30 +192,229 @@ paths:
         '400':
           description: Bad Request. Invalid POST request.
           schema:
             $ref: '#/definitions/BadRequestResponseModel'
         default:
-          description: Requires remote username and password in basic authentication
+          # Without this if any exception that should not occur for the particular API does occur
+          # will be absorbed and won't be returned to user by handleGeneralExceptions method in views/base.py
+          description: Default case for sporadic exceptions like value error or permission denied error or requires remote username and password in basic authentication for posting new data.
           examples:
-            text/html: "Error: REMOTE_USERNAME required"
+            application/json: "{Error: REMOTE_USERNAME required}"
 
+  /rules/{id_or_alias}:
+    get:
+      summary: Returns a rule identified by the id or alias
+      description: >
+        Returns the entire rule identified by the id or alias given in JSON format. [Docs](http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#rules-id-or-alias)
+      tags:
+        - Rules
+      operationId: auslib.web.admin.views.mapper.rules_id_or_alias_get
+      consumes: []
+      produces:
+        - application/json
+        - text/html
+      externalDocs:
+        url: "http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#id2"
+        description: "Returns the entire rule identified by the id or alias given in JSON format."
+      parameters:
+        - name: id_or_alias
+          in: path
+          description: rule_id or alias field value to filter the rule against
+          type: string
+          required: true
 
-definitions:
+      responses:
+        '200':
+          description: Successfully created a new rule
+          schema:
+            $ref: "#/definitions/RulesBaseGET"
+          headers:
+            X-Data-Version:
+              description: data version of the rule
+              type: integer
+              format: int32
+              minimum: 1
+        '404':
+          description: Requested rule did not exist
+          schema:
+            title: Reuested rule not found
+            type: string
+            example: Requested rule does not exist
+    post:
+      summary: Updates the rule
+      description: >
+        Updates the rule identified by the id or alias. [Docs](http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#id3)
+      tags:
+        - Rules
+      operationId: auslib.web.admin.views.mapper.rules_id_or_alias_post
+      consumes: []
+      produces:
+        - application/json
+        - text/html
+      externalDocs:
+        url: "http://mozilla-balrog.readthedocs.io/en/latest/admin_api.html#id2"
+        description: "Returns the entire rule identified by the id or alias given in JSON format."
+      parameters:
+        - name: id_or_alias
+          in: path
+          description: rule_id or alias field value to filter the rule against
```

Comment:
I don't think it makes sense to use the word "filter" here. How about "rule_id or alias to update"?